src/api_crypto.rs

```diff
@@ -302,6 +289,172 @@ impl PublicKey {
     }
 }
 
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct SequesterPrivateKeyDer(pub libparsec::crypto::SequesterPrivateKeyDer);
+
+crate::binding_utils::gen_proto!(SequesterPrivateKeyDer, __repr__);
+
+#[pymethods]
+impl SequesterPrivateKeyDer {
+    #[new]
+    pub fn new(data: &[u8]) -> PyResult<Self> {
+        libparsec::crypto::SequesterPrivateKeyDer::try_from(data)
+            .map(Self)
+            .map_err(|err| PyValueError::new_err(err.to_string()))
+    }
+
+    #[classmethod]
+    fn generate(_cls: &PyType) -> Self {
+        Self(libparsec::crypto::SequesterPrivateKeyDer::default())
+    }
+
+    fn dump<'py>(&self, py: Python<'py>) -> &'py PyBytes {
+        PyBytes::new(py, &self.0.dump())
+    }
+
+    fn dump_pem(&self) -> String {
+        self.0.dump_pem().to_string()
+    }
+
+    #[classmethod]
+    fn load_pem(_cls: &PyType, s: &str) -> PyResult<Self> {
+        libparsec::crypto::SequesterPrivateKeyDer::load_pem(s)
+            .map(Self)
+            .map_err(|err| PyValueError::new_err(err.to_string()))
+    }
+
+    fn decrypt<'py>(&self, py: Python<'py>, data: &[u8]) -> PyResult<&'py PyBytes> {
+        self.0
+            .decrypt(data)
+            .map(|x| PyBytes::new(py, &x))
+            .map_err(|err| CryptoError::new_err(err.to_string()))
+    }
+
+    #[getter]
+    fn public_key(&self) -> SequesterPublicKeyDer {
```

Comment:
Why do we need to be able to convert back and forth between verification key and encryption key ? (and between signature key and decryption key)

A key is supposed to be used only for one need (so only doing verification key or only doing encryption key), so needing this conversion seems very fishy...

Reply:
Done